src/API.jl

```diff
@@ -1277,10 +1278,21 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
     taskwaiting = Set{Base.PkgId}()
     pkgspidlocked = Dict{Base.PkgId,String}()
 
-    function monitor_stderr(pkg, iob)
+    function monitor_stderr(pkg, iob; single_requested_pkg=false)
         try
+            liveprinting = false
             while isopen(iob)
                 str = readline(iob)
```

Comment:
```suggestion
                str = readline(iob; keep=true)
```
(will give better performance)